projects/RabbitMQ.Client/client/impl/ChannelBase.cs

```diff
@@ -1874,7 +1888,7 @@ await tokenRegistration.DisposeAsync()
             }
         }
 
-        private static BasicProperties PopulateActivityAndPropagateTraceId<TProperties>(TProperties basicProperties,
+        private static BasicProperties? PopulateActivityAndPropagateTraceId<TProperties>(TProperties basicProperties,
```

Comment:
I slightly adjusted this avoid creating a new BasicProperty whenever possible and to copy all properties if e.g. the Properties is ReadOnlyProperties